### PR TITLE
Close Loop 1 and fix of 1856287

### PR DESCRIPTION
**Pull request by jyejare**
Automated Bugs:
https://bugzilla.redhat.com/show_bug.cgi?id=1778177
https://bugzilla.redhat.com/show_bug.cgi?id=1778139
https://bugzilla.redhat.com/show_bug.cgi?id=1778181
https://bugzilla.redhat.com/show_bug.cgi?id=1671319

and a fix of:
https://bugzilla.redhat.com/show_bug.cgi?id=1856287 (including default org / loc fixture)

### CODE REVIEW

File: tests/foreman/cli/test_satellitesync.py [hunk 1]
```diff
@@ -409,6 +409,73 @@ def assert_exported_cvv_exists(self, content_view_name, content_view_version):
...
+        cview = ContentView.info(
+            {'name': exporting_cv_name, 'organization-id': self.exporting_org['id']}
+        )
+        default_cvv_id = cview['versions'][0]['id']
+        ContentView.version_export({'export-dir': self.export_dir, 'id': default_cvv_id})
+        exporting_cvv_version = cview['versions'][0]['version']
+        cview_label = cview['label']
+        exported_tar = f'{self.export_dir}/export-{cview_label}-{exporting_cvv_version}.tar'
+        result = ssh.command(f"[ -f {exported_tar} ]")
+        self.assertEqual(result.return_code, 0)
```
Comment (devendra104) [inline tests/foreman/cli/test_satellitesync.py:452]:
```suggestion
        assert result.return_code == 0
```

File: tests/foreman/cli/test_satellitesync.py [hunk 2]
```diff
@@ -409,6 +409,73 @@ def assert_exported_cvv_exists(self, content_view_name, content_view_version):
...
+        exported_packages = Package.list({'content-view-version-id': default_cvv_id})
+        self.assertTrue(len(exported_packages) > 0)
```
Comment (devendra104) [inline tests/foreman/cli/test_satellitesync.py:454]:
```suggestion
        assert len(exported_packages) > 0
```

File: tests/foreman/cli/test_satellitesync.py [hunk 3]
```diff
@@ -409,6 +409,73 @@ def assert_exported_cvv_exists(self, content_view_name, content_view_version):
...
+                'product-id': importing_prod['id'],
+            }
+        )
+        ContentView.version_import(
+            {'export-tar': exported_tar, 'organization-id': importing_org['id']}
+        )
+        importing_cvv = ContentView.info(
+            {'name': importing_cv_name, 'organization-id': importing_org['id']}
+        )['versions']
+        self.assertTrue(len(importing_cvv) >= 1)
```
Comment (devendra104) [inline tests/foreman/cli/test_satellitesync.py:474]:
```suggestion
        assert len(importing_cvv) >= 1
```

File: tests/foreman/cli/test_satellitesync.py [hunk 4]
```diff
@@ -409,6 +409,73 @@ def assert_exported_cvv_exists(self, content_view_name, content_view_version):
...
+        imported_packages = Package.list({'content-view-version-id': importing_cvv[0]['id']})
+        self.assertTrue(len(imported_packages) > 0)
+        self.assertEqual(len(exported_packages), len(imported_packages))
```
Comment (devendra104) [inline tests/foreman/cli/test_satellitesync.py:477]: Same as above.

File: tests/foreman/cli/test_templatesync.py [hunk 5]
```diff
@@ -126,3 +126,25 @@ def test_positive_export_filtered_templates_to_git(self):
...
+
+        :expectedresults: The templates are exported /tmp directory
+
+        :CaseImportance: Medium
+        """
+        dir_path = '/tmp'
+        TemplateSync.exports(
+            {'repo': dir_path, 'organization-id': module_org.id, 'filter': 'ansible'}
+        )
+        assert ssh.command(f'find {dir_path} -type f -name *ansible* | wc -l').stdout[0] == '23'
```
Comment (devendra104) [inline tests/foreman/cli/test_templatesync.py:150]:
```suggestion
        assert ssh.command(f'find {dir_path} -type f -name *ansible*  2>/dev/null| wc -l').stdout[0] == '23'
```

File: tests/foreman/ui/test_templatesync.py [hunk 6]
```diff
@@ -39,6 +39,8 @@ def test_positive_import_templates(session, templates_org, templates_loc):
 
     :id: 524bf384-703f-48a5-95ff-7c1cf97db694
 
+    :bz: 1778181, 1778139
```
Comment (devendra104) [inline tests/foreman/ui/test_templatesync.py:42]: Add the Bugzilla details in the QA whiteboard

File: tests/foreman/cli/test_satellitesync.py (hunk 1 above)
Comment (jyejare) [reply tests/foreman/cli/test_satellitesync.py:452]: I think converting this entire test module to Pytest is a major effort, so I will keep this as it is.

File: tests/foreman/cli/test_satellitesync.py (hunk 2 above)
Comment (jyejare) [reply tests/foreman/cli/test_satellitesync.py:454]: same here

File: tests/foreman/cli/test_satellitesync.py (hunk 3 above)
Comment (jyejare) [reply tests/foreman/cli/test_satellitesync.py:474]: same here

File: tests/foreman/cli/test_satellitesync.py (hunk 4 above)
Comment (jyejare) [reply tests/foreman/cli/test_satellitesync.py:477]: Same as my last long comment.

File: tests/foreman/cli/test_templatesync.py (hunk 5 above)
Comment (jyejare) [reply tests/foreman/cli/test_templatesync.py:150]: We are specially looking into the exact folder `{dir_path}` for modules so I dont think I should redirect the output.

File: tests/foreman/cli/test_templatesync.py [hunk 7]
```diff
@@ -126,3 +126,25 @@ def test_positive_export_filtered_templates_to_git(self):
 
         :CaseAutomation: NotAutomated
         """
+
+    @tier2
+    def test_positive_export_filtered_templates_to_temp_dir(
+        self, module_org, create_import_export_local_dir
```
Comment (jameerpathan111) [inline tests/foreman/cli/test_templatesync.py:132]: @jyejare is `create_import_export_local_dir` required for this test?

File: tests/foreman/ui/test_templatesync.py (hunk 6 above)
Comment (jyejare) [reply tests/foreman/ui/test_templatesync.py:42]: Done!

Comment (jyejare) [reply tests/foreman/ui/test_templatesync.py:42]: Added for all BZs here!

File: tests/foreman/cli/test_templatesync.py (hunk 7 above)
Comment (jyejare) [reply tests/foreman/cli/test_templatesync.py:132]: @jameerpathan111 Not really, thanks for catching that. Now fixed !

File: tests/foreman/cli/test_templatesync.py (hunk 5 above)
Comment (devendra104) [reply tests/foreman/cli/test_templatesync.py:148]: I think it would help to avoid the unnecessary error, It's my suggestion only.

**Review by devendra104 — APPROVED** [review]
ACK

File: tests/foreman/ui/test_templatesync.py [hunk 8]
```diff
@@ -76,48 +81,55 @@ def test_positive_import_templates(session, templates_org, templates_loc):
...
+        assert pt['template']['locked'] is True
         assert pt['locations']['resources']['assigned'][0] == templates_loc.name
         assert pt['organizations']['resources']['assigned'][0] == templates_org.name
         assert f'name: {import_template}' in pt['template']['template_editor']['editor']
 
 
 @tier2
 @upgrade
-def test_positive_export_templates(session):
+def test_positive_export_templates(session, create_import_export_local_dir):
```
Comment (jameerpathan111) [inline tests/foreman/ui/test_templatesync.py:92]:
```suggestion
def test_positive_export_templates(session):
```
forgot to mention this one.

File: tests/foreman/cli/test_templatesync.py (hunk 7 above)
Comment (jameerpathan111) [reply tests/foreman/cli/test_templatesync.py]: thanks, forgot to mention the other test, otherwise it LGTM.

**Review by jameerpathan111 — APPROVED** [review]
ACK

File: tests/foreman/cli/test_templatesync.py (hunk 5 above)
Comment (jyejare) [reply tests/foreman/cli/test_templatesync.py:148]: Yes! But I confirm there wont be any error and if its is I want to see it and failing the test! :)

File: tests/foreman/ui/test_templatesync.py (hunk 8 above)
Comment (jyejare) [reply tests/foreman/ui/test_templatesync.py]: Resolved!

**Review by devendra104 — APPROVED** [review]
ACK

**Pull request merged**